src/tablecloth/time/derived.clj

```diff
@@ -0,0 +1,25 @@
+(ns tablecloth.time.derived)
+
+(defn add-derived
+  "Add metadata derived from a few columns."
+  [ds column-names id data]
+  (-> ds
+      (vary-meta assoc-in
+                 [:derived id]
+                 {:column-names column-names
+                  :columns      (select-keys ds column-names)
+                  :data         data})))
```

Comment:
Per our discussion, perhaps `data` could be `thing` or `metadata`? I also wonder if `id` could be more specific. In fact it is the name of the metadata?

Reply:
Thanks @ezmiller !

Renamed a few things, what do you think?